### ISSUE TITLE
Add nu_plugin_from_arrow 

### ISSUE THREAD

**Issue by GTrunSec** [fdncred/nu_plugin_parquet#3]
# Feature request.
sorry about this issue, it might not be the right place.

we already know that the nushell its data-frame is based on the `Apache arrow` format. So Could we parse the arrow directly from the nushell like parquet plugin?

Thanks for your maintaining work.

**Comment by fdncred** [comment]
I believe if you use nushell compiled with dataframe you can do `to arrow` and `open file.arrow`. Nushell doesn't use dataframes by default and they work a bit differently than regular nushell since they're column based. Someone could also augment this plugin to support the arrow format too. We could call it `from dataframe` or something like that and let it figure out if it's parquet or arrow based on the extension.

**Comment by GTrunSec** [comment]
> Someone could also augment this plugin to support the arrow format too. We could call it from dataframe or something like that and let it figure out if it's parquet or arrow based on the extension.

Yeh, I think we should keep a specific command for parsing that format such as `from arrow/ from parquet, etc`, and the `dataframe` can be a general argument to parse one of the formats by extension. it's also covered most use cases. 

Especially, parsing the data from an output instead of `open .file.format': 

![0:1: tmux  -  NixOS   2022-12-17 16-51-21](https://user-images.githubusercontent.com/21156405/208271862-dbc813b3-357c-4157-aa5c-6b458db62e12.jpg)

**Comment by fdncred** [comment]
ya, i'm fine with whatever but my point was that we probably shouldn't have a plugin called `nu_plugin_from_parquet` that handles a variety of formats. So, we can have a separate plugin named `nu_plugin_from_arrow` or we can combine `from parquet` and `from arrow` in one plugin named something like `nu_plugin_from_dataframe` or some such name that is generic and can contain a variety of dataframe formats.

**Comment by GTrunSec** [comment]
>  we can combine from parquet and from arrow in one plugin named something like nu_plugin_from_dataframe. or some such name that is generic and can contain a variety of data frame formats.

I would agree with that. More make sense. thanks

**Comment by mavam** [comment]
(Chiming in here from the perspective of VAST, as shown in the screenshot, which outputs Arrow. )

It seems like the discussion is about single vs. multi-schema data. Arrow's IPC streaming format allows for interleaving record batches of different schemas, requiring the client to demultiplex them.

NB: I looked the The Book but couldn't find much documentation on this type of data. Is there any?

One way systems do this is treat each stream independently, and Arrow Flight creates a gRPC experience around that. Then you don't have to implement that demultiplexing yourself. But demultiplexing a raw Arrow IPC stream might be something to consider, as gRPC is a bit heavy-weight.

I'd be curious what the state of thinking for such data streams is in the Nu world.

**Comment by fdncred** [comment]
> Arrow's IPC streaming format allows for interleaving record batches of different schemas, requiring the client to demultiplex them.

nushell used to handle things like this by just streaming a new table when the headers changed. I'm not sure it handles it any longer though.

Such a thing could be written though but it just depends on what we want the output to look like. It kind of presupposes that the data will be collected until a certain point and the output in the various schemas.

I'm not a big fan of grpc. We used to have a plugin protocol called cap'n proto, which i believe is a grpc derivative. it was just a pain to deal with and we removed it in favor of the msgpack protocol.

**Comment by mavam** [comment]
> Such a thing could be written though but it just depends on what we want the output to look like. It kind of presupposes that the data will be collected until a certain point and the output in the various schemas.

Right, that's the hard part. The terminal is "single-pass" and simply not well-suited for this. Basically we would want `table`, but "tabbed" or "paged". NDJSON is the next best thing that we have in the real world for heterogeneous structures, but it's not the format we'd want here.

As you said, it either requires buffering up to N events, and then render one `table` per schema, or do something custom that supports interleaved data. I haven't come across yet something that shines in the latter case.
